File: Object-C/SPPage/View/SPPageTagView.m
//
//  SPMainTagView.m
//  Radio
//
//  Created by WealongCai on 15/12/10.
//  Copyright (c) 2015年 qzone. All rights reserved.
//

#import "SPPageTagView.h"


@interface SPPageTagView()
@property (nonatomic, assign, readwrite) BOOL isHighlighted;
@end


@implementation SPPageTagView

- (void)highlightTagView
{
    NSException *exception = [NSException exceptionWithName:@"Method no override Exception" reason:@"Method highlightTagView must be override!" userInfo:nil];
    @throw exception;
}

- (void)unhighlightTagView
{
    NSException *exception = [NSException exceptionWithName:@"Method no override Exception" reason:@"Method unhighlightTagView must be override!" userInfo:nil];
    @throw exception;
}


@end



@implementation SPPageTagTitleView

- (instancetype)initWithFrame:(CGRect)frame
{
    if (self = [super initWithFrame:frame]) {
        self.backgroundColor = [UIColor clearColor];
        self.frame = frame;
        
        self.title = [UILabel new];
        self.title.backgroundColor = [UIColor clearColor];
        
        self.title.textAlignment = NSTextAlignmentCenter;
        self.title.frame = self.bounds;
        
        
        [self addSubview:self.title];
    }
    return self;
}



- (void)setHighlightedTitleColor:(UIColor *)highlightedTitleColor
{
    [super setHighlightedTitleColor:highlightedTitleColor];
}

- (void)setNormalTitleColor:(UIColor *)normalTitleColor
{
    [super setNormalTitleColor:normalTitleColor];
}


- (void)highlightTagView
{
    self.title.textColor = self.highlightedTitleColor;
    self.isHighlighted = YES;
}

- (void)unhighlightTagView
{
    self.title.textColor = self.normalTitleColor ;
    self.isHighlighted = NO;
}

- (void)layoutSubviews
{
    [super layoutSubviews];
    [_title sizeToFit];
    _title.center = CGPointMake(self.frame.size.width/2, self.frame.size.height/2);
}

@end

File: Swift/SPPage/View/SPPageTagView.swift
//
//  SPPageTagView.swift
//  SPPage
//
//  Created by GodDan on 2018/3/10.
//  Copyright © 2018年 GodDan. All rights reserved.
//

import UIKit

class SPPageTagTitleView: UIControl {
    
    private var highlightedTitleColor:UIColor = UIColor.clear
    private var normalTitleColor:UIColor = UIColor.clear
    var titleLable:UILabel = UILabel.init()

    
    convenience init(frame: CGRect, highlightedTitleColor:UIColor, normalTitleColor:UIColor)  {
        self.init(frame: frame)
        self.highlightedTitleColor = highlightedTitleColor
        self.normalTitleColor = normalTitleColor
        
    }
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        self.setup()
        self.frame = frame
        self.titleLable.frame = self.bounds
        
    }
    
    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
        self.setup()
    }
    
    func setup()  {
        self.backgroundColor = UIColor.clear
        self.titleLable.backgroundColor = UIColor.clear
        self.titleLable.textAlignment = NSTextAlignment.center
        self.addSubview(self.titleLable)
    }
    
    func highlightTagView() {
        self.titleLable.textColor = self.highlightedTitleColor
    }
    
    func unHighlightTagView () {
        self.titleLable.textColor = self.normalTitleColor
    }
    
}
